### DISCUSSION TITLE
First compile fails at .#AudioConfig.h: no such file or directory

### DISCUSSION THREAD

**Discussion by cytan299** [pschatzmann/arduino-audio-tools#907] [Q&A] [answered]
Hi,
  This is my first compile of the example given in your wiki (Generating Test Tone https://github.com/pschatzmann/arduino-audio-tools/wiki/Introduction). I compiled to this point and got the error:

`Compiling libraries...

Compiling library "audio-tools"
Using library audio-tools at version 0.9.6 in folder: /Users/cytan/Documents/Arduino/libraries/arduino-audio-tools 
Using library WiFiClientSecure at version 2.0.0 in folder: /Users/cytan/Documents/Arduino/hardware/espressif/esp32/libraries/WiFiClientSecure 
Using library WiFi at version 2.0.0 in folder: /Users/cytan/Documents/Arduino/hardware/espressif/esp32/libraries/WiFi 
stat /Users/cytan/Documents/Arduino/libraries/arduino-audio-tools/src/.#AudioConfig.h: no such file or directory

Compilation error: stat /Users/cytan/Documents/Arduino/libraries/arduino-audio-tools/src/.#AudioConfig.h: no such file or directory`

Of course, the .#AudioConfig.h file is in that directory as a symbolic link to something that doesn't exist:

`~/Documents/Arduino/libraries/arduino-audio-tools/src@belle% ls -la
total 48
drwxr-xr-x  20 cytan  staff    640 Jun 23 20:07 ./
lrwxr-xr-x   1 cytan  staff     44 Jun 23 20:07 .#AudioConfig.h@ -> cytan@Cheng-Yangs-MacBook-Pro-154.local.1687
drwxr-xr-x  15 cytan  staff    480 Jun 23 19:05 ../
drwxr-xr-x   8 cytan  staff    256 Jun 23 19:05 AudioAnalog/
drwxr-xr-x  12 cytan  staff    384 Jun 23 19:05 AudioBasic/
drwxr-xr-x  38 cytan  staff   1216 Jun 23 19:05 AudioCodecs/
-rw-r--r--   1 cytan  staff  14858 Jun 23 19:05 AudioConfig.h
drwxr-xr-x   9 cytan  staff    288 Jun 23 19:05 AudioEffects/
drwxr-xr-x   4 cytan  staff    128 Jun 23 19:05 AudioFilter/
drwxr-xr-x  16 cytan  staff    512 Jun 23 19:05 AudioHttp/
drwxr-xr-x  12 cytan  staff    384 Jun 23 19:05 AudioI2S/
drwxr-xr-x  46 cytan  staff   1472 Jun 23 19:05 AudioLibs/
-rw-r--r--   1 cytan  staff    151 Jun 23 19:05 AudioLogger.h
drwxr-xr-x   8 cytan  staff    256 Jun 23 19:05 AudioMetaData/
drwxr-xr-x  10 cytan  staff    320 Jun 23 19:05 AudioPWM/
drwxr-xr-x  11 cytan  staff    352 Jun 23 19:05 AudioTimer/
drwxr-xr-x  24 cytan  staff    768 Jun 23 19:05 AudioTools/
-rw-r--r--@  1 cytan  staff   2389 Jun 23 19:05 AudioTools.h
drwxr-xr-x   9 cytan  staff    288 Jun 23 19:05 Experiments/
drwxr-xr-x   6 cytan  staff    192 Jun 23 19:05 Video/
~/Documents/Arduino/libraries/arduino-audio-tools/src@belle% less .\#AudioConfig.h
.#AudioConfig.h: No such file or directory

`
Clearly, the symbolic linked file doesn't exist (I assume it's trying to create ports on my Mac). Is there a way to fix this in the Arduino IDE or something else?

Thanks for your help!

cytan

**Comment by pschatzmann** [comment, marked as answer]
Not sure what you did wrong and I can't reproduce your issue: For sure there is no include file called "#AudioConfig.h"!

The complete working example can be found [here](https://github.com/pschatzmann/arduino-audio-tools/blob/main/examples/examples-stream/streams-generator-serial/streams-generator-serial.ino)

**Comment by cytan299** [comment]
Hi PS,
   Duh, I simply removed the file and the compile/link went fine. Very strange that this odd ball file was created because I did a git clone to install.

 Now that  I've successfully compiled and ran the streams-generator-serial.ino on my ESP32-S2 I'd like to use the I2C protocol to stream to its DAC pins.
  Is 
     streams-generator-analog.ino
the correct program to try?

Also, note that I'm using the ESP32-S2 which means that the DAC pins are at GPIO 17 and GPIO 18 and not at GPIO 25 and GPIO 26. Is there any defines I need to do or everything should be automagic?

cytan

**Comment by pschatzmann** [comment]
I2C protocol ? Did you mean I2S: I2C can't be used for streaming. 
If you intend to use an external dac, just read the corresponding Wiki.

ESP32-S2 does not support any I2S output to the internal DAC!

**Comment by cytan299** [comment]
Thanks for telling me that. Well, it sux that ESP32-S2 does not support I2S to its internal DAC. So, which example should I modify to send audio to the internal DAC?

**Comment by pschatzmann** [comment]
No example: you have to use an external DAC!
PWM output however might work as well, if you limit your sample rate,  but I never tested this on the S2...

**Comment by cytan299** [comment]
Thanks for the quick reply. I'll try bit banging which is, of course, the most inefficient way to test the internal DAC with sound. Then I'll think about whether to use ESP32-S2 or not for my project.

Thanks again!

cytan
